woocommerce/woocommerce-gutenberg-products-block
# Simplify Product Collection query attributes by removing attributes which are never used

Simplify Product Collection query attributes by removing attributes (`sticky` and `parents`) that are never used.

## What

Remove unused attributes from Product Collection query attribute.

## Why

Considering we're planning to improve and extend the query attribute it makes sense to clean it up first.

In this PR I'm removing `sticky` and `parents` attributes which:
- are only initiated
- don't have visual representations in filters
- merchants can't change their values
- we never change their values and just pass them further.

## Testing Instructions



_Please consider any edge cases this change may have, and also other areas of the product this may impact._


### Some testing

1. Go to Editor
2. Add Product Collection block
3. Smoke test it (Editor and Frontend)

### Migration 

1. Go to Editor
2. Add Products (Beta) block
3. Disable "Inherit query from template" option and apply some filters
4. Click on "Upgrade to Product Collection" block
5. Expected: Block had upgraded preserving the filters and displays correctly.



### Below Testing instructions are for dev team exclusively

1. On `trunk` branch
2. Go to Editor
3. Add Product Collection
4. Disable "Inherit query from template" option and apply some filters
5. Save the template/post/page
6. Switch to this branch (`clean/simplify-product-collection-attributes`)
7. Make sure the Product Collection blocks added before switching to this branch work correctly, can be modified etc.

## WooCommerce Visibility


Required:


## Checklist

Required:

Conditional:

## Changelog
